test/flint/fmpz-test.jl

```diff
@@ -343,36 +343,37 @@ end
 @testset "fmpz.powering..." begin
    a = fmpz(-12)
 
-   @test a^5 == -248832
+   @test a^5 == a^fmpz(5) == -248832
 
-   @test a^1 == a
-   @test a^1 !== a
+   @test isone(a^0) && isone(a^fmpz(0))
 
-   @test isone(a^0)
+   a = fmpz(2)
+   @test_throws InexactError a^(a^200)
 
    for a in fmpz.(-5:5)
       for e = -5:-1
          if a != 1 && a != -1
             @test_throws DomainError a^e
+            @test_throws DomainError a^fmpz(e)
          end
       end
-      @test a^1 == a
-      @test a^1 !== a
+      @test a^1 == a^fmpz(1) == a
+      @test a^1 !== a^fmpz(1) !== a
    end
 
    a = fmpz(1)
    for e = -2:2
-      @test isone(a^e)
-      @test a^e !== a
+      @test isone(a^e) && isone(a^fmpz(e))
+      @test a^e !== a^fmpz(e) !== a
    end
 
    a = fmpz(-1)
    for e = [-3, -1, 1, 3, 5]
-      @test a^e == a
-      @test a^e !== a
+      @test a^e == a^fmpz(e) == a
+      @test a^e !== a^fmpz(e) == a
```

Comment:
The purpose of the !== test is to make sure that ^ always returns a new object. Thus, this line should be the two
```
@test a^e !== a
@test a^fmpz(e) !== a
```

Reply:
Thanks!